File: Unit_tests_python/EX-05-svg_vector_metadata/Bard.py
import xml.etree.ElementTree as ET

def svg_dot_product(svg_path):
 """
 Calculates the dot product of two vectors found within the SVG's metadata.

 Args:
   svg_path: The path to the SVG file.

 Returns:
   The dot product of the two vectors, or -1 if there are fewer than two vectors
   or if they cannot be parsed.
 """

 try:
   # Parse the SVG file
   tree = ET.parse(svg_path)
   root = tree.getroot()

   # Extract vectors from metadata
   vectors = []
   for child in root.find("metadata"):
     if child.tag == "vector":
       try:
         vectors.append([float(x) for x in child.text.split(",")])
       except ValueError:
         return -1

   # Check if there are at least two vectors
   if len(vectors) < 2:
     return -1

   # Calculate dot product
   return sum(a * b for a, b in zip(vectors[0], vectors[1]))

 except Exception:
   return -1

File: Unit_tests_lua/EX-05-svg_vector_metadata/Bard.lua
function svg_dot_product(svg_path)
    -- Load the SVG file using LuaXML
    local xml = require("LuaXML")
    local parsed_xml = xml.load(svg_path)
   
    -- Find the <metadata> element
    local metadata = parsed_xml:find("//metadata")
    if not metadata then
      return -1  -- Metadata not found
    end
   
    -- Extract the vectors from <vector> elements
    local vectors = {}
    for vector_element in metadata:children("vector") do
      local vector_text = vector_element:text()
      local numbers = {}
      for number_str in vector_text:gmatch("%d+%.?%d*") do  -- Match numbers with optional decimals
        table.insert(numbers, tonumber(number_str))
      end
      if #numbers > 0 then
        table.insert(vectors, numbers)
      end
    end
   
    if #vectors < 2 then
      return -1  -- Fewer than two vectors found
    end
   
    -- Calculate the dot product
    local v1 = vectors[1]
    local v2 = vectors[2]
    local dot_product = 0
    for i = 1, #v1 do
      dot_product = dot_product + v1[i] * v2[i]
    end
   
    return dot_product
   end   